dexter/src/test/java/com/karumi/dexter/DexterInstanceTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import com.karumi.dexter.listener.PermissionDeniedResponse;
+import com.karumi.dexter.listener.PermissionGrantedResponse;
+import com.karumi.dexter.listener.PermissionRequest;
+import com.karumi.dexter.listener.multi.MultiplePermissionsListener;
+import com.karumi.dexter.listener.single.PermissionListener;
+import java.util.Collections;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class DexterInstanceTest {
+
+  private static final String ANY_PERMISSION = "noissimrep yna";
+
+  @Mock AndroidPermissionService androidPermissionService;
+  @Mock Context context;
+  @Mock Intent intent;
+  @Mock Activity activity;
+  @Mock MultiplePermissionsListener multiplePermissionsListener;
+  @Mock PermissionListener permissionListener;
+
+  private DexterInstance dexter;
+
+  @Before public void setUp() {
+    MockitoAnnotations.initMocks(this);
```

Comment:
You can use ``@RunWith(MockitoTestRunner.class)`` and remove this call if you want :)